From 3500484de72c05fa19592e6151826c59cafa373f Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Tue, 22 Jan 2013 22:21:03 -0600
Subject: [PATCH 3/3] imx: Add IMX53 AHCI support

Adds device tree node and ahci_platform bits to make AHCI work on mx53 qsb board.

Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/imx53-qsb.dts |    4 ++++
 arch/arm/boot/dts/imx53.dtsi    |    8 ++++++++
 drivers/ata/ahci_platform.c     |    1 +
 3 files changed, 13 insertions(+)

diff --git a/arch/arm/boot/dts/imx53-qsb.dts b/arch/arm/boot/dts/imx53-qsb.dts
index 05cc562..eb03dae 100644
--- a/arch/arm/boot/dts/imx53-qsb.dts
+++ b/arch/arm/boot/dts/imx53-qsb.dts
@@ -255,6 +255,10 @@
 	};
 };
 
+&ahci {
+	status = "okay";
+};
+
 &audmux {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_audmux_1>;
diff --git a/arch/arm/boot/dts/imx53.dtsi b/arch/arm/boot/dts/imx53.dtsi
index d05aa21..9197a74 100644
--- a/arch/arm/boot/dts/imx53.dtsi
+++ b/arch/arm/boot/dts/imx53.dtsi
@@ -732,5 +732,13 @@
 				status = "disabled";
 			};
 		};
+
+		ahci: ahci@0x10000000 { /* AHCI SATA */
+			compatible = "fsl,imx53-ahci";
+			reg = <0x10000000 0x4000>;
+			interrupts = <28>;
+			clocks = <&clks 124>;
+			status = "disabled";
+		};
 	};
 };
diff --git a/drivers/ata/ahci_platform.c b/drivers/ata/ahci_platform.c
index 7d22da2..ba162f8 100644
--- a/drivers/ata/ahci_platform.c
+++ b/drivers/ata/ahci_platform.c
@@ -96,6 +96,7 @@ static struct scsi_host_template ahci_platform_sht = {
 
 static const struct of_device_id ahci_of_match[] = {
 	{ .compatible = "calxeda,hb-ahci",  .data = &ahci_devtype[AHCI],},
+	{ .compatible = "fsl,imx53-ahci",   .data = &ahci_devtype[IMX53_AHCI],},
 	{ .compatible = "fsl,imx6q-ahci",   .data = &ahci_devtype[IMX6Q_AHCI],},
 	{ .compatible = "snps,spear-ahci", },
 	{},
-- 
1.7.10.4

